### PR TITLE
Add top level datasets page

### PR DESCRIPTION
**Pull request by AyushExel**
Add top level datasets page. This is better than nothing but a lot more to be done including graphics and tables

### Summary
📚🖼️🎯

This pull request adds documentation for the dataset formats supported by the Ultralytics framework for different computer vision tasks. It creates new files `docs/dataset/detection.md`, `docs/dataset/pose.md`, `docs/dataset/segmentation.md`, and `docs/dataset/track.md` that describe the dataset formats for detection, pose estimation, segmentation, and multi-object tracking tasks respectively. It also updates the existing file `docs/dataset/cls.md` to document the dataset format for classification tasks. Additionally, it adds a new file `docs/dataset/index.md` that provides an overview and a navigation guide for the dataset documentation. Finally, it modifies the `mkdocs.yml` file to define the navigation structure for the dataset documentation on the website.

> _`docs/dataset` grows_
> _Explaining formats and tasks_
> _Winter of learning_

### Walkthrough
*  Add documentation for dataset formats for different computer vision tasks ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-286007c8ddfa861cf8d88c21762b5bd54a65285b4ec4e497c92bc24485e91f0dL1-R94), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-111fba7d5b119882317b78bfad9834d95f4ce5fc62663cd6eb0f54db02687cbcL1-R102), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-d89b2fa84968cd370f65a25e11da5373b5bb5e150ae9fc9691375feff745a35dL1-R13), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-1092219e9d2cd022ad9d369645a2b18a2105d028c134e3aa664d359ce2b8ddc3R1-R100), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-c5f994f9633e74328dce79743039fb93b5d5729516fd9f83072d6537f487d7cfL1-R98), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-326ea4bb5d2dcd70b965e3a985759ad703a3dd029e65dde1f2ad63341c9e6682R1-R22))
  * Create a new file `docs/dataset/index.md` that provides an overview and a navigation guide for the dataset formats ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-d89b2fa84968cd370f65a25e11da5373b5bb5e150ae9fc9691375feff745a35dL1-R13))
  * Create new files `docs/dataset/cls.md`, `docs/dataset/detection.md`, `docs/dataset/pose.md`, `docs/dataset/segmentation.md`, and `docs/dataset/track.md` that describe the dataset formats for classification, detection, pose estimation, segmentation, and multi-object tracking tasks respectively ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-286007c8ddfa861cf8d88c21762b5bd54a65285b4ec4e497c92bc24485e91f0dL1-R94), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-111fba7d5b119882317b78bfad9834d95f4ce5fc62663cd6eb0f54db02687cbcL1-R102), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-1092219e9d2cd022ad9d369645a2b18a2105d028c134e3aa664d359ce2b8ddc3R1-R100), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-c5f994f9633e74328dce79743039fb93b5d5729516fd9f83072d6537f487d7cfL1-R98), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-326ea4bb5d2dcd70b965e3a985759ad703a3dd029e65dde1f2ad63341c9e6682R1-R22))
  * Explain how to use torchvision datasets for classification tasks in `docs/dataset/cls.md` ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-286007c8ddfa861cf8d88c21762b5bd54a65285b4ec4e497c92bc24485e91f0dL1-R94))
  * Explain how to use custom datasets or convert existing datasets for detection, pose estimation, and segmentation tasks in `docs/dataset/detection.md`, `docs/dataset/pose.md`, and `docs/dataset/segmentation.md` respectively ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-111fba7d5b119882317b78bfad9834d95f4ce5fc62663cd6eb0f54db02687cbcL1-R102), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-1092219e9d2cd022ad9d369645a2b18a2105d028c134e3aa664d359ce2b8ddc3R1-R100), [link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-c5f994f9633e74328dce79743039fb93b5d5729516fd9f83072d6537f487d7cfL1-R98))
  * Explain how to use pre-trained detection, segmentation, or pose models for tracking multiple objects in video sequences in `docs/dataset/track.md` ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-326ea4bb5d2dcd70b965e3a985759ad703a3dd029e65dde1f2ad63341c9e6682R1-R22))
* Update the navigation structure for the dataset documentation in the `mkdocs.yml` file ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2R164-R170))
  * Add a new section for the dataset documentation under the Dataset Format heading ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2R164-R170))
  * Include links to the new files `docs/dataset/index.md`, `docs/dataset/cls.md`, `docs/dataset/detection.md`, `docs/dataset/pose.md`, `docs/dataset/segmentation.md`, and `docs/dataset/track.md` in the navigation structure ([link](https://github.com/ultralytics/ultralytics/pull/2437/files?diff=unified&w=0#diff-98d0f806abc9af24e6a7c545d3d77e8f9ad57643e27211d7a7b896113e420ed2R164-R170))

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
New documentation for dataset formats and usage across different computer vision tasks.

### 📊 Key Changes
- Added documentation for classification dataset format.
- Introduced dataset documentation for object detection, including label and file formats.
- Included new details for instance segmentation and pose estimation dataset formats.
- Outlined planned support for multi-object tracking dataset format.
- Provided examples of YAML configuration files and Python/CLI usage instructions for training models on the defined datasets.

### 🎯 Purpose & Impact
- **Simplifies Training**: Users can now easily understand how to structure their data for training across different tasks. 🎓
- **Enhances Accessibility**: The new docs make it more accessible for new users to use Ultralytics models with their own datasets. 🛠️
- **Prepares for Future Features**: Lays the groundwork for upcoming support in multi-object tracking, signaling continued development and innovation. 🚀

### CODE REVIEW

**Comment by Laughing-q** [comment]
@AyushExel Why we using sub-level in dataset/detection.md but it shows the same level in docs?🤔
image
image

**Comment by Laughing-q** [comment]
@AyushExel ok I updated some description and now all the formats look correct. But the issue above still exists.

**Comment by AyushExel** [comment]
Not sure why that's happening. I'll take a look tonight

**Comment by glenn-jocher** [comment]
@AyushExel wait for this, are you sure we should break down datasets by task? My first thought was we should have a page for each dataset we support. Maybe we should do both?

```
datasets/
  tasks/
    detect
    segment
    ...
  COCO
  VOC
  Argoverse
  ...
```

**Comment by AyushExel** [comment]
Yes we'll definitely need both. Tasks because the format of labels and dataset is different. I have a "supported datasets" section in each task. I guess you'd like to bring it to the top level with tasks?

**Comment by glenn-jocher** [comment]
@AyushExel ok got it. I'll merge this into updates and work on the docs there.

**Comment by glenn-jocher** [comment]
> @AyushExel Why we using sub-level in dataset/detection.md but it shows the same level in docs?🤔 image image

Hey I noticed that also. Markdown in GitHub and in PyCharm seems to render multi-level lists correctly but not in MkDocs...

I had to move iOS and Android subsections into App for the bullet-point list in https://docs.ultralytics.com/hub/ to fix the problem. Let me see if there's another solution.

**Comment by glenn-jocher** [comment]
@Laughing-q @AyushExel seems like we need to use an asterix for sub-section of un-numbered lists, see https://squidfunk.github.io/mkdocs-material/reference/lists/

```
- Nulla et rhoncus turpis. Mauris ultricies elementum leo. Duis efficitur
  accumsan nibh eu mattis. Vivamus tempus velit eros, porttitor placerat nibh
  lacinia sed. Aenean in finibus diam.

    * Duis mollis est eget nibh volutpat, fermentum aliquet dui mollis.
    * Nam vulputate tincidunt fringilla.
    * Nullam dignissim ultrices urna non auctor.

```

**Comment by glenn-jocher** [comment]
Copilot poem's last lines are always like "[RANDOM_SEASON_HERE] of learning" haha

**Comment by glenn-jocher** [comment]
@AyushExel @Laughing-q merged into `updates` to work on the docs there.
